### PR TITLE
fix: Fix type mapping spark

### PR DESCRIPTION
**Pull request by ElliotNguyen68**
**What this PR does / why we need it**: Currently when inference datatype from a spark source, feast not able to infer column of type decimal, this pr aim to turn decimal type column in spark to double feast data type

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by sudohainguyen** [comment]
I prefer explicit type name instead of using regex 🤔 
how many we need to do?

**Comment by ElliotNguyen68** [comment]
> I prefer explicit type name instead of using regex 🤔 how many we need to do?

https://spark.apache.org/docs/3.1.3/api/python/reference/api/pyspark.sql.types.DecimalType.html, I think the number is so many

File: sdk/python/feast/type_map.py [hunk 1]
```diff
@@ -774,6 +776,10 @@ def spark_to_feast_value_type(spark_type_as_str: str) -> ValueType:
         "array<boolean>": ValueType.BOOL_LIST,
         "array<timestamp>": ValueType.UNIX_TIMESTAMP_LIST,
     }
+    decimal_regex_pattern = r"^decimal\([0-9]{1,2},[0-9]{1,2}\)$"
```
Comment (tokoko) [inline sdk/python/feast/type_map.py:779]: startswith("decimal") is probably more appropriate for now if we don't care about the precision.

**Comment by ElliotNguyen68** [comment]
@tokoko , already change

**Comment by tokoko** [comment]
@ElliotNguyen68 Can you add some handling for array of decimals as well while we are at it? just to keep things symmetrical

**Comment by ElliotNguyen68** [comment]
> @ElliotNguyen68 Can you add some handling for array of decimals as well while we are at it? just to keep things symmetrical

Hi @tokoko , i dont get what you mean ?

**Comment by tokoko** [comment]
@ElliotNguyen68 `type_map` contains mapping for spark data types and appropriate array types (`int` and `array<int>` for example). This change would only add support for `decimal` and leave `array<decimal>` unsupported. Maybe add another step for `spark_type_as_str.startswith("array<decimal")`?

**Review by tokoko — APPROVED** [review]
lgtm

**Comment by ElliotNguyen68** [comment]
@HaoXuAI  hope you can merge the pr

**Review by sudohainguyen — APPROVED** [review]
lgtm

**Pull request merged**